Customizing/global/plugins/Services/Repository/RepositoryObject/ReportCouponNew/classes/class.ilObjReportCouponNew.php

```diff
@@ -0,0 +1,186 @@
+<?php
+require_once 'Customizing/global/plugins/Services/Cron/CronHook/ReportMaster/classes/ReportBase/class.ilObjReportBase.php';
+require_once 'Customizing/global/plugins/Services/Cron/CronHook/ReportMaster/classes/ReportBase/class.catSelectableReportTableGUI.php';
+use CaT\TableRelations as TableRelations;
+use CaT\Filter as Filters;
+
+class ilObjReportCouponNew extends ilObjReportBase {
+	
+	protected $online;
+	protected $admin_mode;
+	protected $relevant_parameters = array();
+	protected $gUser;
+	protected $interpreter;
+	
+	public function __construct($a_ref_id = 0) {
+		global $ilUser;
+		$this->gUser = $ilUser;
+		parent::__construct($a_ref_id);
+		$this->gf = new TableRelations\GraphFactory();
+		$this->pf = new Filters\PredicateFactory();
+		$this->tf = new TableRelations\TableFactory($this->pf, $this->gf);
+
+	}
+
+	protected function createLocalReportSettings() {
+		$this->local_report_settings =
+			$this->s_f->reportSettings('rep_robj_rcpn')
+				->addSetting(
+					$this->s_f
+						->settingBool('admin_mode', $this->plugin->txt('admin_mode'))
+					);
+	}
+
+	public function initType() {
+		 $this->setType("xrcn");
+	}
+
+	public function prepareTable(catSelectableReportTableGUI $table) {
+		$table	->defineFieldColumn($this->plugin->txt("code"), 'code', array('code' => $this->fields['c']['coupon_code']))
```

Comment:
Nein, man könnte das wohl auch über den space schauckeln, aber es ging in dem fall schneller...